dask_geopandas/core.py

```diff
@@ -0,0 +1,330 @@
+    divisions : tuple of index values
+        Values along which we partition our blocks on the index
+    """
+
+    _partition_type = geopandas.base.GeoPandasBase
+
+    def __repr__(self):
+        s = "<dask_geopandas.%s | %d tasks | %d npartitions>"
+        return s % (type(self).__name__, len(self.dask), self.npartitions)
+
+    def to_dask_dataframe(self):
+        """Create a dask.dataframe object from a dask_geopandas object"""
+        return self.map_partitions(M.to_pandas)
+
+    @classmethod
+    def _bind_property(cls, attr):
+        """Map property to partitions and bind to class"""
+
+        def prop(self):
+            meta = getattr(self._meta, attr)
+            token = f"{self._name}-{attr}"
+            return self.map_partitions(getattr, attr, token=token, meta=meta)
+
+        doc = getattr(cls._partition_type, attr).__doc__
+        # Insert disclaimer that this is a copied docstring note that
+        # malformed docs will not get the disclaimer (see #4746).
+        if doc:
+            doc = ignore_warning(doc, cls._partition_type, attr)
+        setattr(cls, name, property(fget=prop, doc=doc))
+
+    @classmethod
+    def _bind_elemwise_comparison_method(
+        cls, name, comparison, original, *args, **kwargs
+    ):
+        """ bind comparison method like GeoSeries.contains to this class """
+
+        def meth(self, other, *args, **kwargs):
+            return elemwise(comparison, self, other, *args, **kwargs)
+
+        meth.__name__ = name
+        setattr(cls, name, derived_from(original)(meth))
+
+    @classmethod
+    def _bind_elemwise_operator_method(cls, name, op, original, *args, **kwargs):
+        """ bind operator method like GeoSeries.distance to this class """
+        # name must be explicitly passed for div method whose name is truediv
+        def meth(self, other, *args, **kwargs):
+            meta = _emulate(op, self, other)
+            return map_partitions(
+                op, self, other, meta=meta, enforce_metadata=False, *args, **kwargs
+            )
+
+        meth.__name__ = name
+        setattr(cls, name, derived_from(original)(meth))
+
+    @property
+    def crs(self):
+        """
+        The Coordinate Reference System (CRS) represented as a ``pyproj.CRS``
+        object.
+
+        Returns None if the CRS is not set, and to set the value it
+        :getter: Returns a ``pyproj.CRS`` or None. When setting, the value
+        can be anything accepted by :meth:`pyproj.CRS.from_user_input`,
+        such as an authority string (eg "EPSG:4326") or a WKT string.
+        """
+        return self._meta.crs
+
+    @crs.setter
+    def crs(self, value):
+        """Sets the value of the crs"""
+        new = self.set_crs(value)
+        self._meta = new._meta
+        self.dask = new.dask
+
+    def set_crs(self, value):
+        """Set the value of the crs on a new object"""
+
+        def set_crs(df, crs):
+            df = df.copy(deep=False)
+            df.crs = crs
```

Comment:
We probably want to check that the `crs` is None (not already set), and in that case raise a warning or error (maybe with a keyword to force it). 

Otherwise I think people will accidentally call this assuming their geometries get transformed.

(maybe for a follow-up though; we should also align with a similar method in GeoPandas)

Reply:
This one I think we should do as a follow-up for sure.

Reply:
#3